pkg/engine2/visualizer.go

```diff
@@ -126,25 +127,27 @@ func (e *Engine) GetViewsDag(view View, ctx solution_context.SolutionContext) (c
 			dstTag := e.GetResourceVizTag(string(DataflowView), dst)
 			switch dstTag {
 			case ParentIconTag:
-				hasPath, err := HasParent(topo, ctx, src, dst)
-				if err != nil {
-					errs = errors.Join(errs, err)
-				}
-				if node.Parent.IsZero() && hasPath {
-					node.Parent = dst
-				}
-			case BigIconTag:
-				hasPath, err := HasPath(topo, ctx, src, dst)
-				if err != nil {
-					errs = errors.Join(errs, err)
-				}
-				if hasPath {
-					edge := construct.SimpleEdge{
-						Source: src,
-						Target: dst,
+				if node.Parent.IsZero() {
+					template, err := e.Kb.GetResourceTemplate(dst)
+					if err != nil {
+						errs = errors.Join(errs, err)
+						continue
+					}
+					if collectionutil.Contains(template.Classification.Is, "cluster") ||
+						collectionutil.Contains(template.Classification.Is, "network") {
+						hasPath, err := HasParent(topo, ctx, src, dst)
+						if err != nil {
+							errs = errors.Join(errs, err)
+						}
+						if node.Parent.IsZero() && hasPath {
```

Comment:
`node.Parent.IsZero()` is always true because it's checked in the outer `if`, right?